### ISSUE TITLE
Make in-app touchpoints modal form more accessible

### ISSUE THREAD

**Issue by emyl3** [CDCgov/prime-simplereport#8574]
## Area or Component

In-app touchpoints modal form

## Quick Summary
The in-app touchpoints modal form can be more accessible through the following:
1. When tabbing through the touchpoints modal with a screenreader like VoiceOver (macOS), it is not clear that the "X" for the Touchpoints modal form is the close modal button since it is read as "times." ([This is probably the exact same issue as the one opened in the static site](https://github.com/CDCgov/prime-simplereport-site/issues/789).)
2. Add a `aria-required="true"` attribute for the `<textbox>` input "What do you think about SimpleReport? Please share feedback on anything you like, or how we can improve."

⚠️ This may be initially straightforward to fix. However, if we ever make changes to the touchpoints form (which we copy and paste the form code directly from touchpoints) we will need to figure out a way to ensure this continues to behave in an accessible manner.

### How to Replicate
- Go to the app
- Log in
- Click the "How can we improve SimpleReport?"
- Try using a screenreader to tab through the form to replicate number 1 above.

## Accessibility Violation
[aria-name-missing-incorrect
](https://docs.deque.com/issue-help/1.0.0/en/aria-name-missing-incorrect)
[aria-required-missing
](https://docs.deque.com/issue-help/1.0.0/en/aria-required-missing)

## Action Requested

- Fix the above accessibility issues
- Ensure there is an easy(?-not sure if possible) or at least structured way to maintain these changes if we update the Touchpoints modal in the future

## Screenshots (optional)
<img width="627" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNGdjQ1E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5b383d89ef40b581ca8b72cd0c67ebb185e252ca/Screenshot%202025-03-04%20at%2011.45.48.png" alt="Screenshot 2025-03-04 at 11.45.48.png" />

## Source Code (optional)

[which file or line is the issue attached to?]: #

**Comment by emyl3** [comment]
⚠️ Please address this issue in conjunction with #8573
